### ISSUE TITLE
RFE: Manually-triggered maintenance checks

### ISSUE THREAD

**Issue by Saklad5** [MegaMek/mekhq#4710]
At the moment, MekHQ only allows maintenance checks to be performed when required. While this is a reasonable interpretation of the rules, it can create frustrating situations where a unit that just spent a week in transit breaks down the instant it is deployed in the field due to the maintenance check triggering.

I propose that players be allowed to manually trigger maintenance checks by selecting a unit in the hangar, thereby resetting the clock. As noted above, this would be most relevant when taking advantage of better facilities (such as a DropShip transport bay) shortly before moving to worse conditions (such as a workshop in the field).

This feature relies on [a rules forum question](https://bg.battletech.com/forums/index.php?topic=85485.0) asking whether maintenance time is meant to be subtracted when a check is performed rather than MekHQ's current behavior of subtracting it every single day. If MekHQ's current behavior is actually intended, manual maintenance checks would probably be unbalanced. If it turns out each check subtracts time, it'd make much more sense.

It also relies in part on [this issue](https://github.com/MegaMek/mekhq/issues/4709), specifically the bit about players being able to see when maintenance checks are approaching. Without that, this is a lot more fiddly to use.

**Comment by IllianiCBT** [comment]
I don’t think this should be actioned, unless we introduce limits on how frequently maintenance can be manually rolled. At which point we might as well keep the system as-is.

if you are in a position where you have at least one Elite team, with factory conditions (garrison contract with a high reputation) it would be very easy to just spam manual maintenance until all Units in your force are A-rated. Which is clearly not the intention of the rules.

**Comment by Saklad5** [comment]
> I don’t think this should be actioned, unless we introduce limits on how frequently maintenance can be manually rolled.

That'd be easy: require the maintenance clock to be at least 1 (or some other value) to trigger a check.

**Comment by IllianiCBT** [comment]
If we do that, I’m not sure of the value in adding the tech to support this, if all we’re doing is allowing users to run maintenance a day or two early.

**Comment by Saklad5** [comment]
Again, the main time to do this is on a DropShip shortly before reaching a contract destination. I've been screwed over quite often by having checks occur immediately after deployment, for instance.

**Comment by Saklad5** [comment]
> if you are in a position where you have at least one Elite team, with factory conditions (garrison contract with a high reputation) it would be very easy to just spam manual maintenance until all Units in your force are A-rated. Which is clearly not the intention of the rules.

One Elite team would still only have 480 minutes of normal working time per day, even if we allowed immediate re-checks.

**Comment by IllianiCBT** [comment]
> > if you are in a position where you have at least one Elite team, with factory conditions (garrison contract with a high reputation) it would be very easy to just spam manual maintenance until all Units in your force are A-rated. Which is clearly not the intention of the rules.
> 
> One Elite team would still only have 480 minutes of normal working time per day, even if we allowed immediate re-checks.

The functionality you’re suggesting would require us to waive maintenance time, or else have the full duration of the remaining maintenance time be deducted in one go - which I assume is what you’re requesting. I still think it’s very abusable and goes against the intention of the rules which is that changing quality should be a matter or weeks (or months) rather than the result of spamming manual maintenance checks when it’s convenient.
